saleor/payment/utils.py

```diff
@@ -373,11 +373,14 @@ def validate_gateway_response(response: GatewayResponse):
 
 
 @transaction.atomic
-def _gateway_postprocess(transaction, payment):
+def gateway_postprocess(transaction, payment):
+    if not transaction.is_success:
+        return
+
     transaction_kind = transaction.kind
 
     if transaction_kind in {TransactionKind.CAPTURE, TransactionKind.CONFIRM}:
-        payment.captured_amount += transaction.amount
+        payment.captured_amount += Decimal(transaction.amount)
```

Comment:
Based on `GatewayResponse` `amount` should be in `Decimal` format. If it has a different format then it should be changed during the creation of transaction. Do you agree?
And the second thing. Decimal without quantizing can be tricky.
```
In [18]: Decimal(1.1)
Out[18]: Decimal('1.100000000000000088817841970012523233890533447265625')
```

Reply:
@salwator  waht about this?

Reply:
temporary hack, removed